### PR TITLE
Tutorial & Tour system

### PR DESCRIPTION
**Pull request by ssakone**
For future testing:

- [x] (?) button in the title bar, if we click on it shows the tutorial of the current page
- [x] Automatically show tutorial the first time we open the app, set the boolean option to false afterwards
- [ ] English text / tutorial for Portfolio page 
    * How to enable a coin
    * How to disable a coin
    * How to delete a custom coin
    * How to add a custom coin
    * How to filter the asset array
    * When you enable a coin, you can search BEP-20, ERC-20 and other coins type in the searchbar
- [ ] English text / tutorial for Wallet page 
    * Tx details
    * Withdraw
    * Receive
    * Disable from sidebar
    * Headers explanation
- [ ] English text / tutorial for Pro trading view
    * How works min_volume/max_volume
    * How works the selection of an order
    * How works best orders
    * How works bidirectional orderbook
    * How works buy
    * How works sell
    * How to cancel an order
    * How to see order history
    * How to see trading history
    * How to export trading history
    * How to change orderbook pair
    * How works the layout
- [ ] English text / tutorial for How to create a wallet (only if user click on (?))

Application tour is a new widget to introduce new user about application & how its work

image

automatically open every restart (for test reason)
text need to be update and another page can be add too.

### CODE REVIEW

**Comment by Milerius** [comment]
@ssakone ninja: error: '../../atomic_defi_design/assets/images/tutorial/step2.png', needed by 'src/atomicdex-desktop_autogen/YCDLW3T4OG/qrc_qml.cpp', missing and no known rule to make it
[12:35]

We cannot compile the actual program - @smk762 cannot start the text field if he is not able to compile.

**Comment by smk762** [comment]
yep, tried to build local (CI not working), and got error:
```[ 93%] Built target antara_world_shared_sources
make[3]: *** No rule to make target '../atomic_defi_design/assets/images/tutorial/step2.png', needed by 'src/atomicdex-desktop_autogen/YCDLW3T4OG/qrc_qml.cpp'.  Stop.
make[2]: *** [CMakeFiles/Makefile2:1117: src/CMakeFiles/atomicdex-desktop.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1125: src/CMakeFiles/atomicdex-desktop.dir/rule] Error 2
make: *** [Makefile:430: atomicdex-desktop] Error 2
```

**Comment by Milerius** [comment]
> yep, tried to build local (CI not working), and got error:
> 
> ```
> make[3]: *** No rule to make target '../atomic_defi_design/assets/images/tutorial/step2.png', needed by 'src/atomicdex-desktop_autogen/YCDLW3T4OG/qrc_qml.cpp'.  Stop.
> make[2]: *** [CMakeFiles/Makefile2:1117: src/CMakeFiles/atomicdex-desktop.dir/all] Error 2
> make[1]: *** [CMakeFiles/Makefile2:1125: src/CMakeFiles/atomicdex-desktop.dir/rule] Error 2
> make: *** [Makefile:430: atomicdex-desktop] Error 2
> ```

This is because Step2.png need to be renamed step2.png, linux is case-sensitive.

**Comment by smk762** [comment]
`Automatically show tutorial the first time we open the app, set the boolean option to false afterwards` at least while testing, perhaps there can be a toggle for this in settings?

I also think we need to "dumb it down" to grandma mode, and start at the start - how to create / restore a wallet. Hyperlinks to online stuff can help keep verbosity down (e.g., what are seed words, private keys etc). Also should include "how to addressbook"

**Comment by smk762** [comment]
Here's a starting point:

image
Login screen tooltips - 
New Wallet: `Generate a new address and seed phrase`
Recover Wallet: `Import an existing address via seed phrase or private key`
My Wallets: `Select a wallet name to login`

"Welcome to AtomicDEX -  Step_1" 
```
AtomicDEX Desktop is revolutionary cryptocurrency wallet where you can store your coins and tokens and make peer to peer trades using [Komodo Platform's](https://komodoplatform.com/) [atomic swap protocol](https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html).

An overview of your portfolio is available in the "Dashboard" tab, with a summary of your hodlings fiat price, fiat value and overall percentage.

The "Wallet" tab  offers additional detail for each enabled coin, including transaction history, receiving address (in text and QR code format) and "send to" form for making payments. The "Addressbook" tab lets you save destination addresses for quick and convenient selection when making payments.

The "DEX" tab is what makes AtomicDEX special. Here, you can create buy or sell orders for any enabled coin pair, or review and select existing trades listed in the orderbook. Being a DEX, there is no middle man - you keep sole control of your keys! Click here more info on [how atomic swaps work.](https://cryptoadventure.org/beginners-guide-to-atomic-swaps-solving-the-centralization-problem/)
```

Step_2 
```
Komodo's Atomic Swap protocol has evolved greatly since the code was written for the first swaps in 2014 between tokens on the NXT platform. Today, it is possible to trade between a great many coins including UTXO based (e.g. BTC, DOGE, KMD), Ethereum / ERC-20 tokens, QTUM / QRC-20 tokens, and BNB / BEP-20 tokens. If a compatible token is not in the default installation, you can even add it yourself thru a simple form in the app!

The test Komodo smartchain tokens RICK and MORTY are available via an in wallet faucet so you can practice how to do an atomic swap without needing to add any coins of real value to the wallet.
```

Step_3
```
As you navigate through the app, keep an eye out for the tooltip info icons to get some info about what the buttons do, and features available within AtomicDEX.

If you have any questions, suggestions or other feedback, we'd love to hear all about it in the [Komodo Platform Discord server](
https://komodoplatform.com/discord) or the projects [Github repository](https://github.com/KomodoPlatform/atomicDEX-Desktop/). 
```

**Comment by ssakone** [comment]
> Here's a starting point:
> 
> image
> Login screen tooltips -
> New Wallet: `Generate a new address and seed phrase`
> Recover Wallet: `Import an existing address via seed phrase or private key`
> My Wallets: `Select a wallet name to login`
> 
> "Welcome to AtomicDEX - Step_1"
> 
> ```
> AtomicDEX Desktop is revolutionary cryptocurrency wallet where you can store your coins and tokens and make peer to peer trades using [Komodo Platform's](https://komodoplatform.com/) [atomic swap protocol](https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html).
> 
> An overview of your portfolio is available in the "Dashboard" tab, with a summary of your hodlings fiat price, fiat value and overall percentage.
> 
> The "Wallet" tab  offers additional detail for each enabled coin, including transaction history, receiving address (in text and QR code format) and "send to" form for making payments. The "Addressbook" tab lets you save destination addresses for quick and convenient selection when making payments.
> 
> The "DEX" tab is what makes AtomicDEX special. Here, you can create buy or sell orders for any enabled coin pair, or review and select existing trades listed in the orderbook. Being a DEX, there is no middle man - you keep sole control of your keys! Click here more info on [how atomic swaps work.](https://cryptoadventure.org/beginners-guide-to-atomic-swaps-solving-the-centralization-problem/)
> ```
> 
> Step_2
> 
> ```
> Komodo's Atomic Swap protocol has evolved greatly since the code was written for the first swaps in 2014 between tokens on the NXT platform. Today, it is possible to trade between a great many coins including UTXO based (e.g. BTC, DOGE, KMD), Ethereum / ERC-20 tokens, QTUM / QRC-20 tokens, and BNB / BEP-20 tokens. If a compatible token is not in the default installation, you can even add it yourself thru a simple form in the app!
> 
> The test Komodo smartchain tokens RICK and MORTY are available via an in wallet faucet so you can practice how to do an atomic swap without needing to add any coins of real value to the wallet.
> ```
> 
> Step_3
> 
> ```
> As you navigate through the app, keep an eye out for the tooltip info icons to get some info about what the buttons do, and features available within AtomicDEX.
> 
> If you have any questions, suggestions or other feedback, we'd love to hear all about it in the [Komodo Platform Discord server](
> https://komodoplatform.com/discord) or the projects [Github repository](https://github.com/KomodoPlatform/atomicDEX-Desktop/). 
> ```

@smk762 HTML formatage is supported ;)

**Comment by ssakone** [comment]
> Here's a starting point:
> 
> image
> Login screen tooltips -
> New Wallet: `Generate a new address and seed phrase`
> Recover Wallet: `Import an existing address via seed phrase or private key`
> My Wallets: `Select a wallet name to login`
> 
> "Welcome to AtomicDEX - Step_1"
> 
> ```
> AtomicDEX Desktop is revolutionary cryptocurrency wallet where you can store your coins and tokens and make peer to peer trades using [Komodo Platform's](https://komodoplatform.com/) [atomic swap protocol](https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html).
> 
> An overview of your portfolio is available in the "Dashboard" tab, with a summary of your hodlings fiat price, fiat value and overall percentage.
> 
> The "Wallet" tab  offers additional detail for each enabled coin, including transaction history, receiving address (in text and QR code format) and "send to" form for making payments. The "Addressbook" tab lets you save destination addresses for quick and convenient selection when making payments.
> 
> The "DEX" tab is what makes AtomicDEX special. Here, you can create buy or sell orders for any enabled coin pair, or review and select existing trades listed in the orderbook. Being a DEX, there is no middle man - you keep sole control of your keys! Click here more info on [how atomic swaps work.](https://cryptoadventure.org/beginners-guide-to-atomic-swaps-solving-the-centralization-problem/)
> ```
> 
> Step_2
> 
> ```
> Komodo's Atomic Swap protocol has evolved greatly since the code was written for the first swaps in 2014 between tokens on the NXT platform. Today, it is possible to trade between a great many coins including UTXO based (e.g. BTC, DOGE, KMD), Ethereum / ERC-20 tokens, QTUM / QRC-20 tokens, and BNB / BEP-20 tokens. If a compatible token is not in the default installation, you can even add it yourself thru a simple form in the app!
> 
> The test Komodo smartchain tokens RICK and MORTY are available via an in wallet faucet so you can practice how to do an atomic swap without needing to add any coins of real value to the wallet.
> ```
> 
> Step_3
> 
> ```
> As you navigate through the app, keep an eye out for the tooltip info icons to get some info about what the buttons do, and features available within AtomicDEX.
> 
> If you have any questions, suggestions or other feedback, we'd love to hear all about it in the [Komodo Platform Discord server](
> https://komodoplatform.com/discord) or the projects [Github repository](https://github.com/KomodoPlatform/atomicDEX-Desktop/). 
> ```

Can we reduce little text size
because its little overflow 
image

Max word count (130) without link recommanded.

File: atomic_defi_design/qml/Constants/Tutorials.qml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        description: qsTr(`AtomicDEX Desktop is revolutionary cryptocurrency wallet where you can store your coins and tokens and make peer to peer trades using <a href="https://komodoplatform.com/">Komodo Platform's</a> <a href"https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html">atomic swap protocol</a>.An overview of your portfolio is available in the "Dashboard" tab, with a summary of your hodlings fiat price, fiat value and overall percentage. The "Wallet" tab  offers additional detail for each enabled coin, including transaction history, receiving address (in text and QR code format) and "send to" form for making payments. The "Addressbook" tab lets you save destination addresses for quick and convenient selection when making payments. The "DEX" tab is what makes AtomicDEX special. Here, you can create buy or sell orders for any enabled coin pair, or review and select existing trades listed in the orderbook. Being a DEX, there is no middle man - you keep sole control of your keys! Click here more info on <a href="https://cryptoadventure.org/beginners-guide-to-atomic-swaps-solving-the-centralization-problem/">how atomic swaps work</a> `),
```
Comment (Milerius) [inline atomic_defi_design/qml/Constants/Tutorials.qml:10]: Each time you use AtomicDEX Desktop here, you want to use the app name, such as done in the FAQ

File: atomic_defi_design/qml/Constants/Tutorials.qml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        image: Images.tour_step2,
+        reverse: false},
+    	{name: qsTr(`SMART PORTFOLIO`),
+        image: Images.tour_step1,
+        description: qsTr(`As you navigate through the app, keep an eye out for the tooltip info icons to get some info about what the buttons do, and features available within AtomicDEX. If you have any questions, suggestions or other feedback, we'd love to hear all about it in the <a href="https://komodoplatform.com/discord">Komodo Platform Discord server</a> or the projects <a href="https://github.com/KomodoPlatform/atomicDEX-Desktop">Github repository</a>. `),
```
Comment (Milerius) [inline atomic_defi_design/qml/Constants/Tutorials.qml:19]: Same here, you do not need to specify the discord, you can use the one from the global variable

**Review by Milerius — CHANGES_REQUESTED** [review]

File: atomic_defi_design/qml/Constants/Tutorials.qml (hunk 1 above)
Comment (Milerius) [reply atomic_defi_design/qml/Constants/Tutorials.qml:10]:
```cpp
engine.rootContext()->setContextProperty("atomic_app_name", QString{DEX_NAME});
    engine.rootContext()->setContextProperty("atomic_app_website_url", QString{DEX_WEBSITE_URL});
    engine.rootContext()->setContextProperty("atomic_app_support_url", QString{DEX_SUPPORT_URL});
    engine.rootContext()->setContextProperty("atomic_app_discord_url", QString{DEX_DISCORD_URL});
    engine.rootContext()->setContextProperty("atomic_app_twitter_url", QString{DEX_TWITTER_URL});
    engine.rootContext()->setContextProperty("atomic_app_primary_coin", QString{DEX_PRIMARY_COIN});
    engine.rootContext()->setContextProperty("atomic_app_secondary_coin", QString{DEX_SECOND_PRIMARY_COIN});
    ```

File: atomic_defi_design/qml/Constants/Tutorials.qml (hunk 2 above)
Comment (Milerius) [reply atomic_defi_design/qml/Constants/Tutorials.qml:19]:
```cpp
   engine.rootContext()->setContextProperty("atomic_app_name", QString{DEX_NAME});
    engine.rootContext()->setContextProperty("atomic_app_website_url", QString{DEX_WEBSITE_URL});
    engine.rootContext()->setContextProperty("atomic_app_support_url", QString{DEX_SUPPORT_URL});
    engine.rootContext()->setContextProperty("atomic_app_discord_url", QString{DEX_DISCORD_URL});
    engine.rootContext()->setContextProperty("atomic_app_twitter_url", QString{DEX_TWITTER_URL});
    engine.rootContext()->setContextProperty("atomic_app_primary_coin", QString{DEX_PRIMARY_COIN});
    engine.rootContext()->setContextProperty("atomic_app_secondary_coin", QString{DEX_SECOND_PRIMARY_COIN});
```

**Comment by smk762** [comment]
Am I correct to assume this will be something like 2-3 pages of intro, then a page for each tab/screen with arrow pointers to widgets etc explaining their use?

Also ignore tooltips for now, can review using them sparingly later (re:chat with @milerius yesterday)

Rewording last post, less verbose - 

### Step 1 - 

<b>AtomicDEX Desktop</b> is revolutionary cryptocurrency wallet where you can hodl your coins and make peer to peer trades with <a href="https://komodoplatform.com/">Komodo Platform's</a> <a href="https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html">atomic swap protocol</a>.

The <b>Dashboard</b> tab shows an overview of your portfolio with a summary of your wallet's coin balances, fiat value and percentage breakdown.

The <b>Wallet</b> shows more detail for enabled coins, like transaction history, receiving address (text/QR) and a form for sending funds.

The <b>Addressbook</b> tab lets you save destination addresses for quick and convenient selection when making payments.

The <b>DEX</b> tab is <a href="https://cryptoadventure.org/beginners-guide-to-atomic-swaps-solving-the-centralization-problem/">what makes AtomicDEX special</a>. You can create trades for any enabled coin pair or select trades from the orderbook. Being a DEX, there is no middle man - you control your keys!


### Step 2

Komodo's <a href="https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html">Atomic Swap protocol</a> has evolved greatly since the code was written for the first atomic swaps in 2014 between tokens on the NXT platform and the <a href ="https://www.youtube.com/watch?v=bf371VQRh0M">first GUI for <b>cross</b> chain atomic swaps (BarterDEX)</a> in 2017.

Now with AtomicDEX, we can do <a href="https://blockmanity.com/news/feature/the-top-3-cross-chain-protocols/">cross-protocol</a> atomic swaps between hundreds of coins including UTXO based (e.g. BTC, DOGE, KMD), Ethereum / ERC-20 tokens, QTUM / QRC-20 tokens, and BNB / BEP-20 tokens. If a compatible token is not yet listed, you can even add it yourself via a wizard in the app!

Free <a href="https://developers.komodoplatform.com/basic-docs/antara/antara-setup/antara-customizations.html">Antara Smartchain</a> tokens RICK and MORTY</a> are available from an in-wallet faucet so you can <a href="https://komodoplatform.com/en/blog/atomicdex-funding-trading/"> practice doing atomic swaps</a> without needing to add other coins to the wallet.

Questions, suggestions or feedback? We'd love to hear all about it in the <a href="https://komodoplatform.com/discord">Komodo Platform Discord server</a> or the AtomicDEX <a href="https://github.com/KomodoPlatform/atomicDEX-Desktop/">Github repository</a>!


### Step 3
<coming soon!>
----------------------------------------------------------------------------------------------------------------------------------

Please let me know if text is succinct enough or if I need to remove more chars.

For the next few pages, seems the simplest option would be to create an image [like what I did for Agama](https://support.komodoplatform.com/support/solutions/articles/29000024495-export-private-key-wif-from-a-wallet-seed-or-pin-password-from-agama-wallet-deprecated-
)  way back when. Optionally, to enhance the aesthetic, we could apply some kind of mouseover highlighting (fade to black/white all elements not under cursor, and full color for widget and its explanation under the cursor).

**Comment by Milerius** [comment]
> Am I correct to assume this will be something like 2-3 pages of intro, then a page for each tab/screen with arrow pointers to widgets etc explaining their use?
> 
> Also ignore tooltips for now, can review using them sparingly later (re:chat with @Milerius yesterday)
> 
> Rewording last post, less verbose -
> 
> ### Step 1 -
> AtomicDEX Desktop is revolutionary cryptocurrency wallet where you can hodl your coins and make peer to peer trades with [Komodo Platform's](https://komodoplatform.com/) [atomic swap protocol](https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html).
> 
> The Dashboard tab shows an overview of your portfolio with a summary of your wallet's coin balances, fiat value and percentage breakdown.
> 
> The Wallet shows more detail for enabled coins, like transaction history, receiving address (text/QR) and a form for sending funds.
> 
> The Addressbook tab lets you save destination addresses for quick and convenient selection when making payments.
> 
> The DEX tab is [what makes AtomicDEX special](https://cryptoadventure.org/beginners-guide-to-atomic-swaps-solving-the-centralization-problem/). You can create trades for any enabled coin pair or select trades from the orderbook. Being a DEX, there is no middle man - you control your keys!
> 
> ### Step 2
> Komodo's [Atomic Swap protocol](https://developers.komodoplatform.com/basic-docs/atomicdex/atomicdex-api.html) has evolved greatly since the code was written for the first atomic swaps in 2014 between tokens on the NXT platform and the [first GUI for cross chain atomic swaps (BarterDEX)](https://www.youtube.com/watch?v=bf371VQRh0M) in 2017.
> 
> Now with AtomicDEX, we can do [cross-protocol](https://blockmanity.com/news/feature/the-top-3-cross-chain-protocols/) atomic swaps between hundreds of coins including UTXO based (e.g. BTC, DOGE, KMD), Ethereum / ERC-20 tokens, QTUM / QRC-20 tokens, and BNB / BEP-20 tokens. If a compatible token is not yet listed, you can even add it yourself via a wizard in the app!
> 
> Free [Antara Smartchain](https://developers.komodoplatform.com/basic-docs/antara/antara-setup/antara-customizations.html) tokens RICK and MORTY are available from an in-wallet faucet so you can [ practice doing atomic swaps](https://komodoplatform.com/en/blog/atomicdex-funding-trading/) without needing to add other coins to the wallet.
> 
> Questions, suggestions or feedback? We'd love to hear all about it in the [Komodo Platform Discord server](https://komodoplatform.com/discord) or the AtomicDEX [Github repository](https://github.com/KomodoPlatform/atomicDEX-Desktop/)!
> 
> ### Step 3
> ## <coming soon!>
> Please let me know if text is succinct enough or if I need to remove more chars.
> 
> For the next few pages, seems the simplest option would be to create an image [like what I did for Agama](https://support.komodoplatform.com/support/solutions/articles/29000024495-export-private-key-wif-from-a-wallet-seed-or-pin-password-from-agama-wallet-deprecated-) way back when. Optionally, to enhance the aesthetic, we could apply some kind of mouseover highlighting (fade to black/white all elements not under cursor, and full color for widget and its explanation under the cursor).

image

Is this way ok for you ?

**Comment by smk762** [comment]
That looks sleek!
Will review wording :+1:

File: atomic_defi_design/qml/Constants/Tutorials.qml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        image: Images.tour_step1,
+        reverse: true},
+    	{name: qsTr(`LOT OF CRYPTO 200+`),
```
Comment (smk762) [inline atomic_defi_design/qml/Constants/Tutorials.qml:13]: LOTS (not LOT)

**Comment by tonymorony** [comment]
@ssakone @smk762 could you please update on what is missing now in this PR to un-draft and merge?

**Comment by smk762** [comment]
@ssakone if you can link me to images (like https://user-images.githubusercontent.com/21139416/119782626-b4bfb300-becc-11eb-8e61-18bbee85899c.png) for review, I can tweak the text as required.
Where images not yet created for items in https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/941#issue-651987316 , I can generate some text but it may need to be refined for clarity / space depending on how the box highlights etc are intended.

**Comment by Milerius** [comment]
This pr is outdated and depends on #951 in order to point widget position in tutorial

**Comment by smk762** [comment]
Eyeball icon is open in some panes and closed in others, does not change on click and no relation to whether panel is visible or hidden (except the "Place Order" panel it looks a bit bolder when shut)
image

**Comment by smk762** [comment]
@tonymorony @SylEze time to close this one in favour of guides / videos within documentation website?
